subsystems/intakesubsystem.py

```diff
@@ -0,0 +1,172 @@
+        Holding = auto()  # arm in feeding position, motor holding
+        Feeding = auto()  # arm in feeding position, motor passing through
+        Staging = auto()  # arm is in position to score amp/trap, motor holding
+        Amp = auto()  # arm in same staging position, motor ejects
+        Trap = auto()  # arm pushes into trap, delay(?), motor ejects
+        Eject = auto()  # arm is down, motor ejects
+
+    def __init__(self) -> None:
+        Subsystem.__init__(self)
+        self.setName(__class__.__name__)  # basic subsystem boilerplate
+
+        self.pivotEncoder = CTREEncoder(
+            constants.kPivotEncoderID,
+            constants.kIntakeAngleOffset.degrees(),
+        )
+
+        self.intakeMotor = NEOBrushless(
+            constants.kIntakeCANID,
+            constants.kIntakeName,
+            constants.kIntakePIDSlot,
+            constants.kIntakePGain,
+            constants.kIntakeIGain,
+            constants.kIntakeDGain,
+            constants.kIntakeInverted,
+        )
+        self.pivotMotor = Talon(
+            constants.kPivotCANID,
+            constants.kPivotName,
+            constants.kPivotPGain,
+            constants.kPivotIGain,
+            constants.kPivotDGain,
+            constants.kPivotInverted,
+        )
+
+        # pivot motor spins 60 times per arm revolution
+        # use absolute encoder to determine motor position
+        pivotMotorPosition = (
+            self.pivotEncoder.getPosition().radians()
+            / constants.kRadiansPerRevolution
+            * constants.kPivotGearRatio
+        )
+        self.pivotMotor.setEncoderPosition(pivotMotorPosition)
+
+        self.state = self.IntakeState.Holding
+
+    def periodic(self) -> None:
+        # get actual velocity values for intake motor later
+
+        if self.state == self.IntakeState.Intaking:
+            self.pivotMotor.set(
+                Talon.ControlMode.Position,
+                constants.kFloorPositionAngle.radians()
+                / constants.kRadiansPerRevolution
+                * constants.kPivotGearRatio,
+            )
+            if self.intakeMotor.getLimitSwitch(NEOBrushless.LimitSwitch.Forwards):
+                self.intakeMotor.set(
+                    NEOBrushless.ControlMode.Position,
+                    self.intakeMotor.get(NEOBrushless.ControlMode.Position),
+                )
+            else:
+                self.intakeMotor.set(
+                    NEOBrushless.ControlMode.Velocity, constants.kIntakeSpeed
+                )
+
+        elif self.state == self.IntakeState.Holding:
+            self.pivotMotor.set(
+                Talon.ControlMode.Position,
+                constants.kHandoffAngle.radians()
+                / constants.kRadiansPerRevolution
+                * constants.kPivotGearRatio,
+            )
+            self.intakeMotor.set(
+                NEOBrushless.ControlMode.Position,
+                self.intakeMotor.get(NEOBrushless.ControlMode.Position),
```

Comment:
this wont work - all it says it "set the position of where it already is at"
take the instance where it moved, it's just going to continue to do nothing since its at the position its at

do some proper state bits

Reply:
Why can't this just be like velocity = 0 or go to break mode or smthing, seems odd to position control it